src/server/load_statistics/linux.rs

```diff
@@ -56,29 +59,45 @@ impl ThreadLoadStatistics {
     /// the usage curve more smooth.
     pub fn record(&mut self, instant: Instant) {
         self.instants[self.cur_pos] = instant;
-        self.cpu_usages[self.cur_pos] = 0f64;
+        self.cpu_usages[self.cur_pos].clear();
         for tid in &self.tids {
             // TODO: if monitored threads exited and restarted then, we should update `self.tids`.
             if let Ok(stat) = thread::full_thread_stat(self.pid, *tid) {
-                self.cpu_usages[self.cur_pos] += thread::linux::cpu_total(&stat);
+                let total = thread::linux::cpu_total(&stat);
+                self.cpu_usages[self.cur_pos].insert(*tid, total);
             }
         }
         let current_instant = self.instants[self.cur_pos];
-        let current_cpu_usage = self.cpu_usages[self.cur_pos];
+        let current_cpu_usages = &self.cpu_usages[self.cur_pos];
 
         let next_pos = (self.cur_pos + 1) % self.slots;
         let earlist_instant = self.instants[next_pos];
-        let earlist_cpu_usage = self.cpu_usages[next_pos];
+        let earlist_cpu_usages = &self.cpu_usages[next_pos];
         self.cur_pos = next_pos;
```

Comment:
I don't read this code before. Seems the calculation of CPU usage is only related to the previous record. So 2 slot num should be enough?

Reply:
It's a sliding window to calculate a smooth usage. /cc @hicqu